src/QueryM/Utxos.purs

```diff
@@ -41,68 +38,41 @@ import QueryM.Ogmios as Ogmios
-  -- require any specific behaviour.
-  utxosAtByWallet address _ = allUtxosAt address
+  utxosAt' :: Ogmios.OgmiosAddress -> QueryM Ogmios.UtxoQR
+  utxosAt' addr' = mkOgmiosRequest Ogmios.queryUtxosAtCall _.utxo addr'
 
-  -- Gets all utxos at an (internal) Address in terms of (internal)
-  -- Cardano.Transaction.Types.
-  allUtxosAt :: Address -> QueryM (Maybe UtxoM)
-  allUtxosAt = addressToOgmiosAddress >>> getUtxos
-    where
-    utxosAt' :: Ogmios.OgmiosAddress -> QueryM Ogmios.UtxoQR
-    utxosAt' addr' = mkOgmiosRequest Ogmios.queryUtxosAtCall _.utxo addr'
+  getUtxos :: Ogmios.OgmiosAddress -> QueryM (Maybe UtxoM)
+  getUtxos address = convertUtxos <$> utxosAt' address
 
-    getUtxos :: Ogmios.OgmiosAddress -> QueryM (Maybe UtxoM)
-    getUtxos address = convertUtxos <$> utxosAt' address
+  convertUtxos :: Ogmios.UtxoQR -> Maybe UtxoM
+  convertUtxos (Ogmios.UtxoQR utxoQueryResult) =
+    let
+      out'
+        :: Array
+             ( Maybe TransactionInput /\ Maybe
+                 TransactionOutput
+             )
+      out' = Map.toUnfoldable utxoQueryResult
+        <#> bimap
+          txOutRefToTransactionInput
+          ogmiosTxOutToTransactionOutput
 
-    convertUtxos :: Ogmios.UtxoQR -> Maybe UtxoM
-    convertUtxos (Ogmios.UtxoQR utxoQueryResult) =
-      let
-        out'
-          :: Array
-               ( Maybe TransactionInput /\ Maybe
-                   TransactionOutput
-               )
-        out' = Map.toUnfoldable utxoQueryResult
-          <#> bimap
-            txOutRefToTransactionInput
-            ogmiosTxOutToTransactionOutput
-
-        out
-          :: Maybe
-               ( Array
-                   ( TransactionInput /\
-                       TransactionOutput
-                   )
-               )
-        out = out' <#> bisequence # sequence
-      in
-        (wrap <<< Map.fromFoldable) <$> out
-
-  -- Nami appear to remove collateral from the utxo set, so we shall do the same.
-  -- This is crucial if we are submitting via Nami. If we decide to submit with
-  -- Ogmios, we can remove this.
-  -- More detail can be found here
-  -- https://github.com/Berry-Pool/nami-wallet/blob/ecb32e39173b28d4a7a85b279a748184d4759f6f/src/api/extension/index.js
-  -- by searching "// exclude collateral input from overall utxo set"
-  -- or functions getUtxos and checkCollateral.
-  namiUtxosAt :: Address -> QueryM (Maybe UtxoM)
-  namiUtxosAt address = getWalletCollateral >>= maybe
-    (liftEffect $ throw "Nami wallet missing collateral")
-    \collateral' -> do
-      let collateral = unwrap collateral'
-      utxos' <- allUtxosAt address
-      pure (over UtxoM (Map.delete collateral.input) <$> utxos')
+      out
+        :: Maybe
+             ( Array
+                 ( TransactionInput /\
+                     TransactionOutput
+                 )
+             )
+      out = out' <#> bisequence # sequence
+    in
+      (wrap <<< Map.fromFoldable) <$> out
```

Comment:
```suggestion
      wrap <<< Map.fromFoldable <$> out
```

Reply:
Let's make sure that we can actually remove Nami-specific filtering of utxos. I know the comment said that it would be possible after switching to Omgios for submission, but I don't know if that's actually true (in fact I suspect it might not be). We might also end up needing to filter out the collateral for Gero as well

Reply:
In fact, considering Nami's collateral mechanism has been added to CIP-30, we might need to filter out the collateral utxos for Gero as well